index.ts

```diff
@@ -1,114 +1,134 @@
-    remove() : void;
+    element(element: d3.Selection<any>): Tip<T>;
+    show(data: T): Tip<T>;
+    hide(): Tip<T>;
+    remove(): void;
 }
 
 export interface TipFactory<T> extends Function {
-    () : Tip<T>
+    (): Tip<T>;
 }
 
-interface WindowDimensions {
-    width : number,
-    height : number
+interface Dimensions {
+    width: number;
+    height: number;
 }
 
 interface Position {
-    top : number,
-    left : number,
+    top: number,
+    left: number,
 }
 
-let windowDimensions : WindowDimensions = {
-    width: window.innerWidth,
-    height: window.innerHeight
-}
-
-function setWindowDimmensions() : void {
-    windowDimensions = {
-        width: window.innerWidth,
-        height: window.innerHeight
-    }
-}
-
-const windowResize = window.addEventListener('resize', setWindowDimmensions);
-
-function getOffset(event : MouseEvent, bounds : ClientRect, offset : Position) : Position {
-    const collideVertically : boolean = (windowDimensions.height + window.scrollY) - event.pageY - offset.top - bounds.height < 0;
-    const collideHorizontally : boolean = (windowDimensions.width + window.scrollX) - event.pageX - offset.left - bounds.width < 0;
+const defaultOffset: Position = { top: 10, left: 10 };
+function getOffsetSettings(offset ?: Offset): Position {
+    if (!offset) { return defaultOffset; }
 
     return {
-        top: collideVertically ? event.pageY - bounds.height - offset.top : offset.top + event.pageY,
-        left: collideHorizontally ? event.pageX - bounds.width - offset.left : event.pageX + offset.left
+        top: offset.top === undefined ? defaultOffset.top : offset.top,
+        left: offset.left === undefined ? defaultOffset.left : offset.left
     };
 }
 
-const defaultOffset : Position = { top: 10, left: 10 };
-function getOffsetSettings(offset? : Offset) : Position {
-    if (!offset) return defaultOffset;
+function getDimensions(el): Dimensions {
     return {
-        top: offset.top === undefined ? defaultOffset.top : offset.top,
-        left: offset.left === undefined ? defaultOffset.left : offset.left
+        width: el.innerWidth,
+        height: el.innerHeight
     };
 }
 
-export default function d3scription<T> (contentGetter : ContentGetter<T>, options:Options = {}) : TipFactory<T> {
-    const offsetSettings : Position = getOffsetSettings(options.offset);
+export default function d3scription<T> (contentGetter: ContentGetter<T>, options: Options = {}): TipFactory<T> {
+    if (typeof window === 'undefined') {
+        return (): Tip<T> => {
+            const methods = {
+                element(element: d3.Selection<any>): Tip<T> {
+                    return methods;
+                },
+                show(data: T): Tip<T> {
+                    return methods;
+                },
+                hide(): Tip<T> {
+                    return methods;
+                },
+                remove(): void {}
+            };
+
+            return methods;
+        };
+    }
```

Comment:
Unfortunately it results in exporting d3scription under `default` key, because initially there are more imports in source file. In typescript it works fine, but when you want to use it in browser: `d3scription` is not a function, `d3scription.default` is.